sources/packages/backend/libs/utilities/src/parallel-processing-utils.ts

```diff
@@ -23,14 +23,27 @@ export enum ParallelIntensity {
  * @param createPromise Implementation that calls processing function which
  * returns a promise.
  * @param inputs Input data from which the bulk processing is done.
- * @param maxParallelRequests Maximum number of parallel processes allowed to be executed at same time.
+ * @param options.
+ * - `maxParallelRequests` maximum number of parallel processes allowed
+ * to be executed at same time.
+ * - `progress` reports the current progress of the parallel processing.
+ * The `currentRecord` parameter reports the number of records processed (not the index).
+ * The method optionally allow to process async operations.
+ * - `partialResults` results from the last processed batch. Every time the promises are
+ * awaited this method is called. The method optionally allow to process async operations.
  * @returns resolved responses of promise.
  */
 export const processInParallel = async <P, I>(
   createPromise: (input: I) => Promise<P>,
```

Comment:
Can we update the signature of create promise to `createPromise: (input: I, ...args: any[]) => Promise<P>`  to state it can accept more args to match the way it has been used now.?